### PR TITLE
Verification key NULLable; topo sort tokens

### PR DESCRIPTION
**Pull request by psteckler**
In the archive db schema, allow the verification key in zkApp accounts to be NULLable; it's an option in the OCaml code.

When adding the tokens used in a block to the db, topologically sort them, so that if token B's owner is an account id with token A, add token A first.

There were related issues with the Berkeley testnet that these changes attempt to fix.

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

File: src/app/archive/lib/processor.ml [hunk 1]
```diff
@@ -3547,12 +3547,49 @@ let add_block_aux ?(retries = 3) ~logger ~pool ~add_block ~hash
...
+        ~f:(fun ((token_id, acct_id_opt) as token_and_owner) ->
+          match acct_id_opt with
+          | None ->
+              assert (Token_id.equal token_id Token_id.default) ;
+              (* no edge to default token *)
+              None
+          | Some acct_id -> (
+              let token_id' = Account_id.token_id acct_id in
+              match
+                List.find tokens_used ~f:(fun (tokid, _) ->
```
Comment (mrmr1993) [inline src/app/archive/lib/processor.ml:3568]: Nit: might be more efficient to build a map once at the beginning and lookup over that (`O(n log n)` vs `O(n^2)`)

**Review by mrmr1993 — APPROVED** [review]

**Review by MartinMinkov — APPROVED** [review]

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Review by deepthiskumar — APPROVED** [review]

**Comment by mitschabaude** [comment]
I don't think we should merge this PR as long as the integration test fails!

**Comment by psteckler** [comment]
!ci-build-me

**Review by nholland94 — APPROVED** [review]

**Pull request merged**